### PR TITLE
Add AggregatedLogListener and change consolelog to sync listener

### PR DESCRIPTION
**Pull request by vicancy**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Common/Loggers/AggregatedLogListener.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+namespace Microsoft.DocAsCode.Common
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+#if NetCore
+    using AggregatedList = System.Collections.Generic.SortedDictionary<LogLevel, System.Collections.Generic.List<ILogItem>>;
+#else
+    using AggregatedList = System.Collections.Generic.SortedDictionary<LogLevel, System.Collections.Generic.List<ILogItem>>;
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/Loggers/AggregatedLogListener.cs]: same?

File: src/Microsoft.DocAsCode.Common/Loggers/AggregatedLogListener.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                default:
+                    return BuildStatus.Succeed;
+            }
+        }
+
+        private enum BuildStatus
+        {
+            Failed,
+            SucceedWithWarning,
+            Succeed
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Common/Loggers/AggregatedLogListener.cs]: >Succeed [](start = 12, length = 7)

minor: compared with "Failed", is "Succeeded" better?

File: src/Microsoft.DocAsCode.Common/Loggers/AggregatedLogListener.cs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+            _innerListener.Flush();
+            WriteFooter(buildStatus);
+            foreach (var level in _aggregatedList.Keys.ToList())
+            {
+                _aggregatedList[level] = new List<ILogItem>();
+            }
+        }
+
+        public void Dispose()
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Common/Loggers/AggregatedLogListener.cs]: >Dispose [](start = 20, length = 7)

_innerListener.Dispose()?

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**